### PR TITLE
tools: dnstester for arm64

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
As mentioned [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/1411#issuecomment-1462359438) we need to create arm64 images for dnstester. This PR reworks the image to handle multi platform support. The corresponding arm64 [jobs](https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/4376025214/jobs/7661323843) seems to be passing fine now.

### CODE REVIEW

**Comment by mauriciovasquezbernal** [comment]
What do you think about adding a job in github actions to automatically update this image if changed? We can use https://github.com/dorny/paths-filter to only trigger this when the relevant files (tools/dnstester/*) are changed. 

It can be seem as some work upfront but in the long term I think that'd save us some headaches.

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -416,6 +416,39 @@ jobs:
...
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build ${{ matrix.image.name }} image
+      uses: docker/build-push-action@v4
+      with:
+        context: ${{ matrix.image.dockerfile-dir }}
+        file: ${{ matrix.image.dockerfile-dir }}/Dockerfile
+        push: true
+        tags: ghcr.io/inspektor-gadget/${{ matrix.image.name }}:${{ matrix.image.tag }}
+        platforms: linux/amd64,linux/arm64
```
Comment (mauriciovasquezbernal) [inline .github/workflows/inspektor-gadget.yml:450]: nit: probably this should be also in the matrix as I suspect we're going to push other images that don't need to be multiplatform using this action in the future, like ebpf-builder.

**Comment by mqasimsarfraz** [comment]
> What do you think about adding a job in github actions to automatically update this image if changed? We can use https://github.com/dorny/paths-filter to only trigger this when the relevant files (tools/dnstester/*) are changed.
> 
> It can be seem as some work upfront but in the long term I think that'd save us some headaches.

Thanks for https://github.com/dorny/paths-filter, I included the filter. 

Also, my idea is to still push images for PRs with different tags to allow testing if needed and only do a push to `latest` when changes are merged. Does that make sense?

**Comment by mauriciovasquezbernal** [comment]
> Also, my idea is to still push images for PRs with different tags to allow testing if needed and only do a push to latest when changes are merged. Does that make sense?

I don't have a strong opinion on this one. I see you already implemented this so I think we can keep this behavior. From time to time we could perform some manual cleanup of those tags but it's not so important as those images are intended for internal usage.

**Review by mauriciovasquezbernal — APPROVED** [review]
I see the run succeeded in https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/4384930120/jobs/7677008837 and the image was pushed https://github.com/inspektor-gadget/inspektor-gadget/pkgs/container/dnstester/76425713?tag=1414-merge. Approving it.

Thanks a lot for handling all the comments.

**Pull request merged**